audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,163 @@
+      let output = event.outputBuffer.getChannelData(i);
+      let weights = this.computeGain(envelope);
+
+      for (let j = 0; j < input.length; j++) {
+        output[j] = weights[j] * input[j];
+      }
+    }
+  }
+
+  /**
+   * Detect level using the difference equation for the Root Mean Squared 
+   * value of the input signal.
+   * @param  {AudioBuffer} inputBuffer input audio buffer
+   * @return {Float32Array} envelope the registered level of the input             
+   */
+  detectLevel(inputBuffer) {
+    let channel = new Float32Array(event.inputBuffer.getChannelData(0).length);
+    // Stereo input is downmixed to mono input via averaging.
+    if (channel.numberOfChannels == 2) {
+      for (let i = 0; i < inputBuffer.getChannelData(0).length; i++) {
+        channel[i] = (inputBuffer.getChannelData(0)[i] +
+                          inputBuffer.getChannelData(1)[i]) / 2;
+      }
+    } else {
+      channel = inputBuffer.getChannelData(0);
+    }
+    // Set the value at each index of the envelope to be a positive value
+    // that combines the current and previous value according to alpha.
+    // See http://www.aes.org/e-lib/browse.cfm?elib=16354 for details.
+    let envelope = new Float32Array(channel.length);  
+    envelope[0] = this.alpha_ * this.lastLevel +
+        (1 - this.alpha_) * Math.pow(channel[0], 2);
+
+    for (let j = 1; j < channel.length; j++) {
+      envelope[j] = this.alpha_ * envelope[j - 1] +
+          (1 - this.alpha_) * Math.pow(channel[j], 2);
+    }
+    this.lastLevel = envelope[envelope.length - 1];
+    // Scale the envelope levels back to the original magnitude.
+    for (let k = 0; k < envelope.length; k++) {
+      envelope[k] = Math.sqrt(envelope[k]) * (1 / 0.7);
+    }
+    return envelope;
+  }
+
+  /**
+   * Computes an array of weights which determines what samples are silenced.
+   * @param  {Float32Array} envelope the level of the input
+   * @return {Float32Array} weights numbers in the range 0 to 1 set in 
+   *                                accordance with the threshold, the envelope,
+   *                                and attack and release                             
+   */
+  computeGain(envelope) {
+    let weights = new Float32Array(envelope.length);
+    // When attack or release are 0, the weight changes between 0 and 1
+    // in one step.
+    let attackSteps = 1;
+    let releaseSteps = 1;
+    let attackLossPerStep = 1;
+    let releaseGainPerStep = 1;
+    // When attack or release are > 0, the associated weight changes between 0 
+    // and 1 in the number of steps corresponding to the millisecond attack 
+    // or release time parameters.
+    if (this.attack > 0) {
+      attackSteps = Math.ceil(this.context_.sampleRate * this.attack);
+      attackLossPerStep = 1 / attackSteps;  
+    }
+    if (this.release > 0) {
+      releaseSteps = Math.ceil(this.context_.sampleRate * this.release);
+      releaseGainPerStep = 1 / releaseSteps;
+    }
```

Comment:
Looks like you linearly adjust the weights for attack and release?  Is this what you want? Maybe adjust in dB units?  (Just asking)